app/com/arpnetworking/rollups/QueueActor.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2020 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.rollups;
+
+import akka.actor.AbstractActor;
+import akka.actor.Props;
+import akka.japi.pf.ReceiveBuilder;
+import com.arpnetworking.logback.annotations.Loggable;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.ArrayDeque;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Queue;
+
+/**
+ * An actor with a bounded queue.
+ *
+ * @param <T>
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public class QueueActor<T extends Serializable> extends AbstractActor {
```

Comment:
Should this be a de-duping queue?

Reply:
A good point! I've generalized this to a `CollectionActor`, which in practice is backed by a set.